torch/_inductor/scheduler.py

```diff
@@ -1525,15 +1525,15 @@ def add_user(used_by_name, user_node, can_inplace=False, is_weak=False):
 
         # make sure input mutation isn't dead-code-eliminated
         for name in self.mutation_renames:
-            if name in V.graph.graph_inputs:
+            if name in V.graph.graph_inputs or name in V.graph.mutated_buffers:
                 add_user(name, OutputNode(StarDep(name)))
                 V.graph.mutated_inputs.add(name)
 
         inp_names = {
             name: index for index, name in enumerate(V.graph.graph_inputs.keys())
         }
         V.graph.mutated_input_idxs = [
-            inp_names[name] for name in V.graph.mutated_inputs
+            inp_names[name] for name in V.graph.mutated_inputs if name in inp_names
```

Comment:
While this does fix the issue, I think it is creating something a bit odd.

`V.graph.mutated_inputs` will now contain non-inputs and also not match with `V.graph.mutated_input_idxs`.

Similarly, we now have non-ouputs with a dependency on `OutputNode`.


These types of things (where variables names don't reflect their actual contents) will make the codebase harder to work in.

Can we fix the underlying issue here properly?  Perhaps we can just add a check for `name not in V.graph.mutated_buffers` to the dead code elimination pass?

Reply:
As an alternative solution, could we normalize module attributes to be inputs to inductor's graph? This would allow inductor to distinguish between mutated intermediate buffers that actually are dead code, and mutated attributes which are observable.